pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,410 @@
+type PodSandboxListItem struct {
+	ID    PodSandboxID
+	State PodSandboxState
+	// Labels are key value pairs that may be used to scope and select individual resources.
+	Labels Labels
+}
+
+// PodSandboxStatus contains the status of the PodSandbox.
+type PodSandboxStatus struct {
+	// ID of the sandbox.
+	ID PodSandboxID
+	// State of the sandbox.
+	State PodSandboxState
+	// Network contains network status if network is handled by the runtime.
+	Network *PodSandboxNetworkStatus
+	// Status specific to a Linux sandbox.
+	Linux *LinuxPodSandboxStatus
+	// Labels are key value pairs that may be used to scope and select individual resources.
+	Labels Labels
+	// Annotations is an unstructured key value map.
+	Annotations map[string]string
+}
+
+// PodSandboxNetworkStatus is the status of the network for a PodSandbox.
+type PodSandboxNetworkStatus struct {
+	IPs []string
+}
+
+// Namespaces contains paths to the namespaces.
+type Namespaces struct {
+	// Network is the path to the network namespace.
+	Network string
+}
+
+// LinuxSandBoxStatus contains status specific to Linux sandboxes.
+type LinuxPodSandboxStatus struct {
+	// Namespaces contains paths to the sandbox's namespaces.
+	Namespaces *Namespaces
+}
+
+// PodSandboxResources contains the CPU/memory resource requirements.
+type PodSandboxResources struct {
+	// CPU resource requirement.
+	CPU resource.Quantity
+	// Memory resource requirement.
+	Memory resource.Quantity
+}
+
+// This is to distinguish with existing ContainerID type, which includes a
+// runtime type prefix (e.g., docker://). We may rename this later.
+type RawContainerID string
+
+// ContainerRuntime provides methods for container lifecycle operations, as
+// well as listing or inspecting existing containers. These methods should
+// either return an error or block until the operation succeeds.
+type ContainerRuntime interface {
+	// Create creates a container in the sandbox, and returns the ID
+	// of the created container.
+	Create(config *ContainerConfig, sandboxConfig *PodSandboxConfig, sandboxID PodSandboxID) (RawContainerID, error)
+	// Start starts a created container.
+	Start(id RawContainerID) error
+	// Stop stops a running container with a grace period (i.e., timeout).
+	Stop(id RawContainerID, timeout int) error
+	// Remove removes the container.
+	Remove(id RawContainerID) error
+	// List lists the existing containers that match the ContainerFilter.
+	// The returned list should only include containers previously created
+	// by this ContainerManager.
+	List(filter ContainerFilter) ([]ContainerListItem, error)
+	// Status returns the status of the container.
+	Status(id RawContainerID) (RawContainerStatus, error)
+	// Exec executes a command in the container.
+	Exec(id RawContainerID, cmd []string, streamOpts StreamOptions) error
+}
+
+// ContainerListItem provides the runtime information for a container returned
+// by List().
+type ContainerListItem struct {
```

Comment:
It doesn't have all the information. If we want to call it `Container`, we should probably include the config as well.